index.js

```diff
@@ -144,15 +163,23 @@ function URL(address, location, parser) {
  *
  * @param {String} prop Property we need to adjust.
  * @param {Mixed} value The newly assigned value.
+ * @param {Boolean|Function} fnOrNoSlashes When setting the query, it will be the
+ *                                         function used to parse the query.
+ *                                         When setting the protocol, double slash will
+ *                                         be removed from the final url if it is true.
  * @returns {URL}
  * @api public
  */
-URL.prototype.set = function set(part, value, fn) {
+URL.prototype.set = function set(part, value, fnOrNoSlashes) {
   var url = this;
 
   if ('query' === part) {
+    if (fnOrNoSlashes && 'function' !== typeof fnOrNoSlashes) {
```

Comment:
I would not add this, it will throw an error anyway if the given argument is not function.
It will save few bytes on the browser.